index.rst

```diff
@@ -1,61 +1,191 @@
 
-  To add images, add the image file (png, svg or jpeg preferred) to the
-  _static/ directory. The reST syntax for adding the image is
+1. **Broadcast messages,** that apply to all users and are visible in a specific time window
+2. **Direct messages,** that are written for specific users in response to user activity
 
-  .. figure:: /_static/filename.ext
-     :name: fig-label
+At launch, Semaphore focuses entirely on the broadcast message use-case, with direct messaging support being added later.
 
-     Caption text.
+The broadcast message use-case is vitally important during |DP01|. :cite:`RTN-004`
+During development, we have relied on Slack to notify users about maintenance events and system outages.
+Since |DP01| users won't be serviced through Slack :cite:`RTN-006`, we need to leverage user-facing channels to begin communicating this information.
+While we could directly use the `Community forum`_, we don't believe that the forum is the right channel for providing timely information to users.
+Instead, messages about things like maintenance events need to be served to users where they are — namely in the |RSP| user interfaces, such as the Squareone_ homepage, the Notebook Aspect, and the Portal Aspect.
+This use case implies the need for a small web service, embedded in the |RSP| deployment, that can serve messages for display on |RSP| user interfaces.
+This service is Semaphore.
 
-   Run: ``make html`` and ``open _build/html/index.html`` to preview your work.
-   See the README at https://github.com/lsst-sqre/lsst-technote-bootstrap or
-   this repo's README for more info.
+.. _stack:
 
-   Feel free to delete this instructional comment.
+Technical stack
+===============
 
-:tocdepth: 1
+Semaphore, with respect to the initial use case of broadcast messaging, consists of these technical components:
 
-.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+- GitHub repository serving as a content management system
+- Semaphore service layer, providing an API
+- UI components embedded in Squareone_, and possibly other |RSP| applications, to consume the Semaphore API and display messages to the user
 
-.. sectnum::
+GitHub makes sense as a CMS because it provides both a user interface and an authorization system (including pull request reviews) that the team is already familiar with.
+Further, the cadence of broadcast messages is slow enough that a GitHub-based process of committing a content change and merging a pull request is reasonably efficient.
+We anticipate that a GitHub repository could be dedicated to |RSP| broadcast messages (or the Phalanx deployment repository could be co-opted for broadcast messages).
+Messages are individual files in a plain text format that supports metadata.
+Two possible formats are YAML or Markdown with YAML-formatted front-matter (the latter option makes it easier for authors to write and format body content).
```

Comment:
Having spent some time experimenting with ways of doing this for another project, I recommend making the message YAML and allowing keys to contain YAML values. Then, when writing the messages, make extensive use of `|`.

I have previously experimented with having a Markdown file alongside a separate YAML file of metadata, and it turned out to be surprisingly irritating to deal with. The advantages of having everything in one file that can be parsed in a single pass are substantial, even if it means people's editors won't necessarily show as nice of an interface for editing the Markdown.

Reply:
That's interesting. So many blog engines are built on Markdown with YAML front matter. I guess it comes down to whether the YAML or the markdown is considered more important? If all of our message bodies are one paragraph (or less), then opting for YAML is certainly the way to go.

Reply:
Yeah, that's a good call. My application had multiple Markdown bits and the overall metadata was probably equally important to the Markdown parts, which isn't as much like a blog post. I think this may look more like my application, though.